src/types.ts

```diff
@@ -1603,7 +1603,9 @@ export type SomeZodObject = ZodObject<
   any
 >;
 
-function deepPartialify(schema: ZodTypeAny): any {
+export function deepPartialify<T extends ZodTypeAny>(
```

Comment:
I think it would be a better to split off this update to `deepPartialify` since it's not necessary for the rest of the work here.